core/network/address.go

```diff
@@ -45,6 +45,20 @@ const (
 	ConfigLoopback AddressConfigType = "loopback"
 )
 
+// IsValidAddressConfigType returns whether the given value is a valid
+// method to configure a link-layer network device's IP address.
+// TODO (manadart 2021-05-04): There is an issue with the usage of this
+// method in state where we have denormalised the config method so it is
+// against device addresses. This is because "manual" indicates a device that
+// has no configuration by default. This could never apply to an address.
+func IsValidAddressConfigType(value string) bool {
```

Comment:
Why not `value AddressConfigType`?

Reply:
Is it for stuff coming over a params type?

Reply:
Since I removed the validation from `machine.SetDevicesAddresses` in state, this is not actually called. I have some shuffling about in the works for config type, and if it remains unused I will remove it.